### PR TITLE
Feat/merge firebase auth with signin composable

### PR DESCRIPTION
**Pull request by davidholzwarth**
# Summary

This PR merges the Firebase Authentication functionality with the SignIn UI. This means that now when pressing the sign in button, the sign in process gets triggered, as opposed to simulating a successful signin. 

The sign in UI was changed, as well as the sign in screen tests and the end to end tests.

# Details

Depends on the nature of the PR :
- **Code changes :** describe the conceptual approach taken in the code and tell the reviewer what part of the changes does what, so that it's easier to read the code
- **Other changes :** Add details if relevant, or remove this section if not useful

This task has been handled by @davidholzwarth for sprint 2 and 3, and by @rBergCode for sprint 4

### CODE REVIEW

**Comment by rBergCode** [comment]
Note: Sonar is failing here and it's normal. I removed tests in order to merge this. It will be fix soon with another PR.

File: app/src/main/java/ch/hikemate/app/ui/auth/SignInScreen.kt [hunk 1]
```diff
@@ -18,32 +23,72 @@ import androidx.compose.material3.ButtonDefaults
...
+  // Create the launcher for adding a Google account in case there is no Google account connected
+  // to the device. Necessary since the Google sign-in process requires a Google account to be
+  // connected to the device, otherwise the sign-in process will fail with a No Credential Exception
+  val addAccountLauncher =
+      rememberLauncherForActivityResult(
+          contract = ActivityResultContracts.StartActivityForResult()) { result: ActivityResult ->
+            Toast.makeText(
+                    context,
+                    "Connected Google Account to your device successfully. Please wait while we retry the signup.",
+                    Toast.LENGTH_LONG)
```
[the remark this answers is not in the document]
Comment (mchac15) [reply app/src/main/java/ch/hikemate/app/ui/auth/SignInScreen.kt:77]: Also the message seems to be too long for the toast and appears cut after some words.

[the remark this answers is not in the document]
Comment (mchac15) [reply app/src/main/java/ch/hikemate/app/ui/auth/SignInScreen.kt:77]: Bug: It still looks like there is an error with Google Authentication whenever I try to put my email to Sign In I get an error message saying: "This account already exists"

**Review by mchac15 — COMMENTED** [review]
Still seems like there is some bugs in the code you need to handle. I don't know if it's a me problem. Please tell me if you don't get the same.

**Review by mchac15 — CHANGES_REQUESTED** [review]
Check the comments I made and get in contact if you need anything

File: app/src/main/java/ch/hikemate/app/ui/auth/SignInScreen.kt [hunk 2]
```diff
@@ -18,32 +23,72 @@ import androidx.compose.material3.ButtonDefaults
...
+
+  // Create the launcher for adding a Google account in case there is no Google account connected
+  // to the device. Necessary since the Google sign-in process requires a Google account to be
+  // connected to the device, otherwise the sign-in process will fail with a No Credential Exception
+  val addAccountLauncher =
+      rememberLauncherForActivityResult(
+          contract = ActivityResultContracts.StartActivityForResult()) { result: ActivityResult ->
+            Toast.makeText(
+                    context,
+                    "Connected Google Account to your device successfully. Please wait while we retry the signup.",
```
[the remark this answers is not in the document]
Comment (rBergCode) [reply app/src/main/java/ch/hikemate/app/ui/auth/SignInScreen.kt:76]: Fixed!

File: app/src/main/java/ch/hikemate/app/ui/auth/SignInScreen.kt (hunk 1 above)
[the remark this answers is not in the document]
Comment (rBergCode) [reply app/src/main/java/ch/hikemate/app/ui/auth/SignInScreen.kt:77]: This should have been fixed. Can you provide me your SH1 key from the task `signingReport` in private ?

**Review by mchac15 — DISMISSED** [review]
LGTM

File: app/src/androidTest/java/ch/hikemate/app/ui/auth/SignInScreenTest.kt [hunk 3]
```diff
@@ -53,7 +55,8 @@ class SignInScreenTest : TestCase() {
 
   @Test
   fun everythingIsOnScreen() {
-    setupSignInScreen()
+      setupSignInScreen()
+    composeTestRule.onNodeWithTag(AppIcon.TEST_TAG).assertIsDisplayed()
     composeTestRule.onNodeWithTag("appIcon").assertIsDisplayed()
```
Comment (mchac15) [inline app/src/androidTest/java/ch/hikemate/app/ui/auth/SignInScreenTest.kt:59]: Duplicate here

**Review by mchac15 — CHANGES_REQUESTED** [review]
One small change but then it's good.

File: app/src/androidTest/java/ch/hikemate/app/ui/auth/SignInScreenTest.kt (hunk 3 above)
Comment (rBergCode) [reply app/src/androidTest/java/ch/hikemate/app/ui/auth/SignInScreenTest.kt:59]: Good catch!

**Review by Wylwi — APPROVED** [review]
LGTM now. Don't forget to put tests back!

Note: @mchac15 did all the review job here. This review is not mine, just approved for timing purposes

**Pull request merged**